Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ChannelGroup/ChannelGroupItemForm.tsx

```diff
@@ -0,0 +1,86 @@
+﻿//******************************************************************************************************
+//  ChannelGroupDetailsForm.tsx - Gbtc
+//
+//  Copyright © 2020, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  07/05/2023 - Parker Dinsdale
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+import * as React from 'react';
+import { SystemCenter } from '@gpa-gemstone/application-typings';
+import { Input, Select } from '@gpa-gemstone/react-forms';
+import { Search } from '@gpa-gemstone/react-interactive';
+import { MeasurmentTypeSlice, MeasurementCharacteristicSlice, ValueListSlice, ValueListGroupSlice } from '../Store/Store';
+import { useAppSelector, useAppDispatch } from '../hooks';
+
+interface IProps {
+    Record: SystemCenter.Types.ChannelGroupDetails,
+    Setter: (record: SystemCenter.Types.ChannelGroupDetails) => void,
+    SetErrors?: (e: string[]) => void
+}
+
+export default function ChannelGroupItemForm(props: IProps) {
+    const dispatch = useAppDispatch();
+
+    const measurementTypeData = useAppSelector(MeasurmentTypeSlice.Data);
+    const measurementTypeStatus = useAppSelector(MeasurmentTypeSlice.Status);
+    const measurementCharacteristicData = useAppSelector(MeasurementCharacteristicSlice.Data);
+    const measurementCharacteristicStatus = useAppSelector(MeasurementCharacteristicSlice.Status);
+    const valueListData = useAppSelector(ValueListSlice.Data);
+    const valueListStatus = useAppSelector(ValueListSlice.Status);
+    const valueListGroupData = useAppSelector(ValueListGroupSlice.Data);
+    const valueListGroupStatus = useAppSelector(ValueListGroupSlice.Status);
+
+    React.useEffect(() => {
```

Comment:
Split these into sepperate uesEffects. We don't need to fire everything every time. React takes care of it.